plugins/com.redhat.fabric8analytics.lsp.eclipse.ui/src/com/redhat/fabric8analytics/lsp/eclipse/ui/EditorComposite.java

```diff
@@ -0,0 +1,145 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat Inc..
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat Incorporated - initial API and implementation
+ *******************************************************************************/
+
+package com.redhat.fabric8analytics.lsp.eclipse.ui;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Collections;
+import java.util.Set;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.equinox.security.storage.StorageException;
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.m2e.editor.pom.MavenPomEditor;
+import org.eclipse.m2e.editor.pom.MavenPomEditorPage;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.browser.Browser;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.ui.IViewPart;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.forms.widgets.FormToolkit;
+import org.json.JSONException;
+
+import com.redhat.fabric8analytics.lsp.eclipse.core.RecommenderAPIException;
+import com.redhat.fabric8analytics.lsp.eclipse.core.RecommenderAPIProvider;
+import com.redhat.fabric8analytics.lsp.eclipse.ui.internal.AnalysesJobHandler;
+import com.redhat.fabric8analytics.lsp.eclipse.ui.internal.Fabric8AnalysisPreferences;
+import com.redhat.fabric8analytics.lsp.eclipse.ui.internal.MessageDialogUtils;
+import com.redhat.fabric8analytics.lsp.eclipse.ui.internal.ThreeScaleIntegration;
+import com.redhat.fabric8analytics.lsp.eclipse.ui.internal.TokenCheck;
+import com.redhat.fabric8analytics.lsp.eclipse.ui.internal.WorkspaceFilesFinder;
+
+/**
+ * Class to create composite for page Fabric8Analyses.
+ * 
+ * @author Geetika Batra
+ *
+ */
+public class EditorComposite extends Composite{
+
+	protected MavenPomEditorPage editorPage;
+
+	public static Browser editorBrowser;
```

Comment:
Is it really safe to have these variables as static? What happens, if you open  2 differennt pom files, won't it be messed up?

Reply:
Removing static from browser.